src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.addAlive();
+  }
+
+  addAlive() {
+    Animal.alive.push(this);
+  }
+
+  healthCheck() {
+    Animal.alive = Animal.alive.filter((item) => item !== this);
+  }
```

Comment:
[CODE PERFORMANCE] The `healthCheck` method is currently removing the animal from the `Animal.alive` array by filtering for all animals except `this`. However, since the method is called when the animal's health is less than or equal to 0, it would be more efficient to filter out all animals whose health is less than or equal to 0 in one go, rather than calling `filter` each time an animal's health reaches zero. Consider refactoring this to improve performance.